sdk/testing-utils/src/mocks/managers/BlockchainClientProviderMock/BlockchainClientProviderMock.ts

```diff
@@ -0,0 +1,24 @@
+import { BlockchainClientProvider, IBlockchainClient } from '@summerfi/blockchain-client-provider'
+import { IConfigurationProvider } from '@summerfi/configuration-provider'
+import { IChainInfo } from '@summerfi/sdk-common'
+
+export class BlockchainClientProviderMock extends BlockchainClientProvider {
+  public readonly rpcUrl: string
+
+  constructor(params: { configProvider: IConfigurationProvider; rpcUrl: string }) {
+    super(params)
+
+    this.rpcUrl = params.rpcUrl
+  }
+
+  public getBlockchainClient(params: { chainInfo: IChainInfo }): IBlockchainClient {
+    return this.getCustomBlockchainClient({ rpcUrl: this.rpcUrl, chainInfo: params.chainInfo })
```

Comment:
As said above I think setting the fork url in transport should be enough

Reply:
As discussed, I'm not sure if the Chain from viem (which also contains the rpc URL) should be also updated. Just in case I'm doing that but I've improved the code to make it more succinct (agreed offline)